libraries/apollo-runtime/src/linuxMain/kotlin/com/apollographql/apollo/network/http/DefaultHttpEngine.kt

```diff
@@ -0,0 +1,51 @@
+package com.apollographql.apollo.network.http
+
+import com.apollographql.apollo.api.http.HttpHeader
+import com.apollographql.apollo.api.http.HttpMethod
+import com.apollographql.apollo.api.http.HttpRequest
+import com.apollographql.apollo.api.http.HttpResponse
+import io.ktor.client.HttpClient
+import io.ktor.client.engine.curl.Curl
+import io.ktor.client.request.request
+import io.ktor.client.request.setBody
+import io.ktor.client.statement.bodyAsChannel
+import io.ktor.util.StringValuesImpl
+import io.ktor.util.flattenEntries
+import io.ktor.utils.io.asSource
+import kotlinx.io.okio.asOkioSource
+import okio.buffer
+
+actual fun DefaultHttpEngine(timeoutMillis: Long): HttpEngine = LinuxHttpEngine()
+
+private class LinuxHttpEngine() : HttpEngine {
+
+  private val client = HttpClient(Curl)
+
+  override suspend fun execute(request: HttpRequest): HttpResponse {
+    val response = client.request(urlString = request.url) {
+      method = when (request.method) {
+        HttpMethod.Get -> io.ktor.http.HttpMethod.Get
+        HttpMethod.Post -> io.ktor.http.HttpMethod.Post
+      }
+      setBody(request.body)
```

Comment:
Pretty sure this is wrong, looking into what is required to map this.

Reply:
Maybe what's in the Ktor adapter [here](https://github.com/apollographql/apollo-kotlin-ktor-support/blob/main/apollo-engine-ktor/src/commonMain/kotlin/com/apollographql/ktor/http/KtorHttpEngine.kt) can be copy/pasted. This would also leave out `kotlinx.io.okio` (not really an issue, just a tendency to avoid a dependency in general).